main.tf

```diff
@@ -304,7 +304,7 @@ data "aws_iam_policy_document" "this" {
         }
       }
 
-      dynamic "condition" {
```

Comment:
@bryantbiggs Sorry, I don't explain clearly. The problem is not naming issue. The element "condition" not mapping to statement value "conditions". if you wana following original document, the next line(308) should be change to like sample.
```
for_each = try(statement.value.condition, [])
```

Reply:
are you saying that this doesn't work as intended? https://github.com/terraform-aws-modules/terraform-aws-opensearch/blob/f4792509f652f8c085cdcdf3c9b576d6588bc8ca/examples/complete/main.tf#L143-L147

Reply:
correct...my sample is workable for now. it needs "condition" and "conditions".
![image](https://github.com/terraform-aws-modules/terraform-aws-opensearch/assets/11845980/f5b61e21-c304-4887-84ec-64f3c6401b88)
because the target is not same....
![image](https://github.com/terraform-aws-modules/terraform-aws-opensearch/assets/11845980/0bc743be-4e42-4aef-84a9-cec30e776b30)